### ISSUE TITLE
Rename folder deleted my notes and folder

### ISSUE THREAD

**Issue by sleekspeed22** [laurent22/joplin#4428]
Rename folder deleted my notes and folder... but not consistently... did this twice within minutes and this happened once

## Environment

Joplin version: Joplin 1.6.8 (prod, win32)
 Client ID: d657549275e24fad82d8fbb7bad62289
 Sync Version: 2
 Profile Version: 34
 Keychain Supported: Yes
 Revision: fc7bd41c1 (master)
Platform: Windows
OS specifics: Windows 10 20H2 Build 19042.746

## Steps to reproduce

1. create nested folders with notes (2 levels deep: Learn > Data Science/Linux/etc ) 
2. rename Learn to Learning
3. hit sync button (I am synching with 2 devices and have fail-safe checked on)

## Describe what you expected to happen
Joplin started to delete the remote notes... no new folder named Learning and notes seem to be gone completely

## Logfile
[log.txt](https://github.com/laurent22/joplin/files/5876492/log.txt)

**Comment by sleekspeed22** [comment]
Luckily I was expecting issues and I exported as JEX file... and I reimported to see if the same rename will delete... well just reimport created 133 remote items... but that Learn folder only has 26 notes with 5 folders. Rename worked as expected this time and only update 1 remote item.  This is very strange and disconcerting as it's not repeatable all the time.  Rename/deleted has happened to me twice in one week of use.
